app/services/umrdr/uniqname_ldap_roles.rb

```diff
@@ -0,0 +1,76 @@
+module Umrdr
+
+  require 'rubygems'
+  require 'net/ldap'
+
+# For local development you can use the https://www.incommon.org/cert/repository/InCommonServerCA.txt
+# as a certiticate for verifying the MCommunity LDAP server's certificate.
+# ENV['LDAPTLS_CACERT'] = '/www/www.lib/misc/bundle.crt';
+
+  class UniqnameLdapRoles
+
+    attr_accessor :uniqname, :role_is_good
+
+    HOST = 'ldap.umich.edu'
+    PORT = 636
+    ENCRYPTION = :simple_tls
+    USERNAME =  'mcommunityuser'
+    PASSWORD = 'mcommunitypass'
+    AUTH = {:method => :simple, :username => USERNAME, :password => PASSWORD}
+
+    def self.uniqname_fac_or_staff(uname)
+      Umrdr::UniqnameLdapRoles.new(uname).run
+    end
+
+    def initialize(uname)
+      @uniqname = uname
+    end
+
+    # Errors returned from ldap.open in run method can include
+    # "getaddrinfo: nodename nor servname provided, or not known" when HOST is wrong
+    # "Operation timed out - user specified timeout" when PORT is wrong
+    # "undefined method `umichinstroles' for #<Net::LDAP::Entry:OBJECT-NUMBER>" when  USERNAME is wrong
+    # "undefined method `umichinstroles' for #<Net::LDAP::Entry:OBJECT-NUMBER>" when  PASSWORD is wrong
+    # "Error: Invalid binding information" when the AUTH is wrong
+
+
+    def test_roles(umichinstroles)
```

Comment:
If this function should return a boolean value, the name should end in `?` and be named in such a way to clue developers into what is being answered.